### PR TITLE
NRT FA dataset

### PR DESCRIPTION
**Pull request by knl88**
Initial setup for daily FA dataset. Will need to check the various metadata fields, but it is a start.

### CODE REVIEW

File: dscreator/sources/ferrybox/uuid_variable_code_mapper.py [hunk 1]
```diff
@@ -14,6 +14,9 @@
         "salinity": "314cd400-14a7-489a-ab97-bce6b11ad068",  # psu, sbe45, fa/ferrybox/sbe45/salinity/raw
         "oxygen": "b345a87f-bdb7-4115-8290-710a1b5264f5",  # rbr, %, fa/ferrybox/rbr/oxygen_saturation/raw
         "track": "4d9ff393-25a3-47b8-aaf1-8fbbccfec3c3",  # fa/gpstrack
+        "chlorophyll": "ea782920-a5e1-415e-930b-b7c12e3cf366", # FA/FERRYBOX/C3/CHLA_FLUORESCENCE/SENSOR_ADJUSTED
+        "turbidity": "de22c594-b313-4c13-ae84-1810eb5841e2", # FA/FERRYBOX/C3/TURBIDITY/SENSOR_ADJUSTED
+        "cdom": "a02eea6b-042d-4209-8307-c88230639b86", # FA/FERRYBOX/C3/CDOM_FLUORESCENCE/SENSOR_ADJUSTED
```
Comment (lisapro) [inline dscreator/sources/ferrybox/uuid_variable_code_mapper.py:19]: usually the shortcut for cdom fluorescence is fdom.

File: dscreator/sources/ferrybox/extractor.py [hunk 2]
```diff
@@ -93,7 +93,7 @@ def first_timestamp(self) -> datetime:
         Padded with 10 sec
         """
```
Comment (lisapro) [inline dscreator/sources/ferrybox/extractor.py:94]: why first timestamp or that date?

File: dscreator/main.py [hunk 3]
```diff
@@ -131,5 +132,36 @@ def sios(max_time_slice: int = 24, stop_after_n_files: int = -1, acdd: bool = Fa
...
+        variable_codes=["temperature", "salinity", "oxygen", "chlorophyll", "turbidity", "cdom"],
+        variable_uuid_map=ferrybox.uuid_variable_code_mapper.MAPPER["FA_20"],
+        qc_flags=[1],
+    )
+
+    dataset_builder = trajectories.ferrybox.DailyFantasy(
+        uuid="no.niva:af11ba01-dfe3-4432-b9d2-4e6fd10714db",
+        dataset_name="nrt_color_fantasy",
+        station_name="color_fantasy",
+        project_name="NorSoop",
```
Comment (lisapro) [inline dscreator/main.py:151]: I guess now the project should be AquaInfra? NorSOOP ended a couple of years ago

File: dscreator/sources/ferrybox/extractor.py (hunk 2 above)
Comment (knl88) [reply dscreator/sources/ferrybox/extractor.py:94]: was a random date to get started but should be updated

File: dscreator/datasets/trajectories/ferrybox.py [hunk 4]
```diff
@@ -117,11 +139,23 @@ def variable_attributes(self, variable_name: str):
...
+class DailyFantasy(NorsoopFantasy):
+    def dataset_attributes(self, ds: xr.Dataset) -> FerryboxDatasetAttrs:
+        attrs = super().dataset_attributes(ds)
+        attrs.title = "Ferrybox on MS Color Fantasy, daily data"
+        attrs.title_no = "Ferje på MS Color Fantasy, daglige data"
+        attrs.summary = "Daily updating dataset of ferrybox data for ferry sailing Oslo, Norway to Kiel, Germany. For more information see: https://www.colorline.no/oslo-kiel/fakta."
+        attrs.summary_no = "Daglig updatert dataset fra ferrybox ombord ferje fra Oslo, Norge til Kiel, Tyskland. For mer informasjon se: https://www.colorline.no/oslo-kiel/fakta."
+        attrs.keywords += "," + ",".join(
+            [
+                "GCMDSK:EARTH SCIENCE > OCEANS > OCEAN CHEMISTRY > CHLOROPHYLL",
```
Comment (lisapro) [inline dscreator/datasets/trajectories/ferrybox.py:157]: They have 
EARTH SCIENCE > OCEANS > OCEAN OPTICS > FLUORESCENCE
Can we create 
EARTH SCIENCE > OCEANS > OCEAN OPTICS > CHLOROPHYLL? 

IN the same way as they have 
EARTH SCIENCE > OCEANS > OCEAN OPTICS > TURBIDITY

File: dscreator/main.py (hunk 3 above)
Comment (knl88) [reply dscreator/main.py:151]: this is just used to specify the storage location, but guess we can update it:)

**Review by lisapro — APPROVED** [review]
I think this is good as a start! but I want to discuss the metadata, I think I have found some errors in metho website (or I dont understand smth)

File: dscreator/datasets/trajectories/ferrybox.py (hunk 4 above)
Comment (knl88) [reply dscreator/datasets/trajectories/ferrybox.py:157]: Can check it comes from here https://gcmd.earthdata.nasa.gov/KeywordViewer

File: dscreator/datasets/trajectories/ferrybox.py [hunk 5]
```diff
@@ -81,10 +81,13 @@ def variable_attributes(self, variable_name: str):
                 return asdict(FlagAttrs(long_name="Sea Water Temperature Quality Flag"))
 
             case "turbidity":
-                return asdict(
+                attrs = asdict(
                     CFVariableAttrs(standard_name="sea_water_turbidity", long_name="Sea Water Turbidity", units="FTU")
                 )
-
+                attrs["ancillary_variables"] = "salinity_qc"
```
Comment (lisapro) [inline dscreator/datasets/trajectories/ferrybox.py:87]: turbidity_qc?

**Comment by knl88** [comment]
@lisapro  would it work to switch to aanderaa for 2022? then we can use https://github.com/NIVANorge/datasetcreator/pull/21/commits/55427e09c507b7af4014bf46b9c2b1454127657c

**Comment by lisapro** [comment]
> @lisapro would it work to switch to aanderaa for 2022? then we can use [55427e0](https://github.com/NIVANorge/datasetcreator/commit/55427e09c507b7af4014bf46b9c2b1454127657c) would also perhaps be a clearer point of change:)

yes, I checked the data and it will be even better as RBR was frozen partly in March 2022, while Anderaa was measuring correctly

File: dscreator/sources/ferrybox/uuid_variable_code_mapper.py [hunk 6]
```diff
@@ -4,17 +4,28 @@
     {
         "temperature": "720b78cb-3e82-4c4d-9b63-7d1ae1b7afc1",  # degC, FA/INLET/SBE38/TEMPERATURE/RAW
         "salinity": "314cd400-14a7-489a-ab97-bce6b11ad068",  # PSU, SBE45, FA/FERRYBOX/SBE45/SALINITY/RAW
-        "oxygen": "e03fba93-1fed-49c2-ac5a-601dc2475915",  # RBR, %, FA/ferrybox/INLET/OPTODE/OXYGEN/SATURATION
+        "oxygen_sat": "e03fba93-1fed-49c2-ac5a-601dc2475915",  # RBR, %, FA/ferrybox/INLET/OPTODE/OXYGEN/SATURATION
```
Comment (lisapro) [inline dscreator/sources/ferrybox/uuid_variable_code_mapper.py:7]: Anderaa

**Pull request merged**